<h1>Activity 1 - PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>
<h3>Analyst : Junnalyn M. Merillo</h3>

In [1]:
#import libraries

import numpy as np
import pandas as pd

<h3>Converting Dataset(s) into DataFrame</h3>
<h2>Requirements</h2>
<ul>
    <li>
        Name of the DataFrame
    </li>
    <li>
        File Path
    </li>
</ul>

In [2]:
penguin = pd.read_csv('Datasets\\penguins.csv')

In [3]:
penguin

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [4]:
penguin.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguin.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [6]:
penguin.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [7]:
penguin.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1>Filtering Column</h1>

In [8]:
penguin['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [9]:
penguin['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [10]:
penguin['species'] =='Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [11]:
penguin[penguin['species'] == 'Chinstrap']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

<h2>Creating Data Frame</h2>

In [12]:
Chinstrap_DF = penguin[penguin['species'] == 'Chinstrap']

In [13]:
Chinstrap_DF

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [14]:
Gentoo_DF = penguin[penguin['species']=='Gentoo']

In [15]:
Gentoo_DF

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [16]:
Adelie_DF = penguin[penguin['species'] == 'Adelie']

In [17]:
Adelie_DF

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [18]:
penguin[(penguin['species']=='Adelie') & (penguin['body_mass_g'] <= 4000)]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
4    Adelie  Torgersen            36.7           19.3              193.0   
5    Adelie  Torgersen            39.3           20.6              190.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
4         3450.0  female  2007  
5         3650.0    male  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[116 rows x 8 columns]

In [19]:
Adelie_BodyMass = penguin[(penguin['species'] == 'Adelie') & (penguin['body_mass_g'] <= 4000)] 

In [20]:
Adelie_BodyMass.info()

<class 'pandas.core.frame.DataFrame'>
Index: 116 entries, 0 to 151
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            116 non-null    object 
 1   island             116 non-null    object 
 2   bill_length_mm     116 non-null    float64
 3   bill_depth_mm      116 non-null    float64
 4   flipper_length_mm  116 non-null    float64
 5   body_mass_g        116 non-null    float64
 6   sex                112 non-null    object 
 7   year               116 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 8.2+ KB


<h2>Filling Up Null Values</h2>

In [21]:
penguin.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [22]:
penguin['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [23]:
penguin['bill_length_mm'] = penguin['bill_length_mm'].fillna(penguin['bill_length_mm'].mean())

In [24]:
penguin.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [25]:
penguin['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [26]:
penguin['bill_depth_mm'] = penguin['bill_depth_mm'].fillna(penguin['bill_depth_mm'].mean())

In [27]:
penguin.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [28]:
penguin['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [29]:
penguin['flipper_length_mm'] = penguin['flipper_length_mm'].fillna(penguin['flipper_length_mm'].mean())

In [30]:
penguin.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [31]:
penguin['body_mass_g'].mean()

np.float64(4201.754385964912)

In [32]:
penguin['body_mass_g'] = penguin['body_mass_g'].fillna(penguin['body_mass_g'].mean())

In [33]:
penguin.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [34]:
penguin['sex'].mode()

0    male
Name: sex, dtype: object

In [35]:
penguin['sex'] = penguin['sex'].fillna(penguin['sex'].mode()[0])

In [36]:
penguin.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [37]:
penguin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


<h1>Changing Data Type</h1>

In [38]:
penguin ['species'] = penguin ['species'].astype('category')
penguin ['island'] = penguin ['island'].astype('category')
penguin ['sex'] = penguin ['sex'].astype('category')
penguin ['year'] = penguin ['year'].astype('category')

In [39]:
penguin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h1>How to create your own column?</h1>

In [40]:
penguin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [41]:
penguin['mass_kg'] = np.round(penguin['body_mass_g'] / 1000, 1)

In [42]:
penguin

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

In [43]:
penguin ['mass_kg'] = penguin ['mass_kg'].astype('float64')

In [44]:
penguin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
 8   mass_kg            344 non-null    float64 
dtypes: category(4), float64(5)
memory usage: 15.4 KB


In [45]:
penguin['mass_kg'].max()

np.float64(6.3)

<h1>Explanatory Data Analysis</h1>
<hr>

<h2>Question 1. Which species has the longest flipper length?</h2>
<hr>

In [46]:
np.round (Chinstrap_DF['flipper_length_mm'].mean())

np.float64(196.0)

In [47]:
np.round (Adelie_DF['flipper_length_mm'].mean())

np.float64(190.0)

In [48]:
np.round (Gentoo_DF['flipper_length_mm'].mean())

np.float64(217.0)

<h3>Insight 1</h3>
<p>Based from the result above, Gentoo penguins has longer flipper length than Chinstrap and Adelie. </p>
<hr>

<h2>Question 2. How many male penguin do Chinstrap have?</h2>
<hr>

In [49]:
((penguin['species'] == 'Chinstrap') & (penguin['sex'] == 'male')).sum()

np.int64(34)

<h3>Insight 2</h3>
<p>Based from the result shown above, there are 34 male penguin under the species of Chinstrap.  </p>
<hr>

<h2> Question 3. How many Adelie penguins are from Dream island?</h2>
<hr>

In [50]:
((penguin['species'] == 'Adelie') & (penguin['island'] == 'Dream')).sum()

np.int64(56)

In [51]:
((penguin['species'] == 'Adelie') & (penguin['island'] == 'Torgersen')).sum()

np.int64(52)

<h3>Insight 3</h3>
<p>Shown above are the results of Adelie penguins coming from Dream and Torgersen island. There are 56 total of Adelie penguins from Dream island and total of 52 Adelie penguins from Torgersen. </p>
<hr>

<h2>Question 4. Which among Chinstrap and Adelie penguins has the heaviest body mass?</h2>
<hr>

In [52]:
np.round (Chinstrap_DF['body_mass_g'].mean())

np.float64(3733.0)

In [53]:
np.round (Adelie_DF['body_mass_g'].mean())

np.float64(3701.0)

<h3>Insight 4</h3>
<p>Based from the result, the maximum body mass of Adelie penguin is 3701 while the Chinstrap is 3733 and is considered with the heaviest body mass. </p>
<hr>

<h2>Question 5. How many Adelie penguins are there in the data from the year 2007?</h2>
<hr>

In [54]:
Adelie_Year = penguin[(penguin['species'] == 'Adelie') & (penguin['year'] == 2007)] 

In [55]:
Adelie_Year.info()

<class 'pandas.core.frame.DataFrame'>
Index: 50 entries, 0 to 49
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            50 non-null     category
 1   island             50 non-null     category
 2   bill_length_mm     50 non-null     float64 
 3   bill_depth_mm      50 non-null     float64 
 4   flipper_length_mm  50 non-null     float64 
 5   body_mass_g        50 non-null     float64 
 6   sex                50 non-null     category
 7   year               50 non-null     category
 8   mass_kg            50 non-null     float64 
dtypes: category(4), float64(5)
memory usage: 3.0 KB


<h3>Insight 5</h3>
<p>Shown above is the index showing that, there are 50 Adelie penguins recorded in the dataset from the year 2007. </p>
<hr>

<h2>Question 6. How many Gentoo penguins have a bill depth of less than 20 mm?</h2>
<hr>

In [56]:
((penguin['species'] == 'Gentoo') & (penguin['bill_depth_mm'] < 20)).sum()

np.int64(124)

<h3>Insight 6</h3>
<p>Based from the result above, there are 124 total of Gentoo penguins that has a bill depth of less than 20 mm. </p>
<hr>

<h2>Question 7. How many Chinstrap and Gentoo penguins are recorded from year 2009?</h2>
<hr>

In [57]:
Chins_09 = penguin[(penguin['species'] == 'Chinstrap') & (penguin['year'] == 2009)] 

In [58]:
Chins_09.info()

<class 'pandas.core.frame.DataFrame'>
Index: 24 entries, 320 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            24 non-null     category
 1   island             24 non-null     category
 2   bill_length_mm     24 non-null     float64 
 3   bill_depth_mm      24 non-null     float64 
 4   flipper_length_mm  24 non-null     float64 
 5   body_mass_g        24 non-null     float64 
 6   sex                24 non-null     category
 7   year               24 non-null     category
 8   mass_kg            24 non-null     float64 
dtypes: category(4), float64(5)
memory usage: 1.7 KB


In [59]:
Gentoo_09 = penguin[(penguin['species'] == 'Gentoo') & (penguin['year'] == 2009)] 

In [60]:
Gentoo_09.info()

<class 'pandas.core.frame.DataFrame'>
Index: 44 entries, 232 to 275
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            44 non-null     category
 1   island             44 non-null     category
 2   bill_length_mm     44 non-null     float64 
 3   bill_depth_mm      44 non-null     float64 
 4   flipper_length_mm  44 non-null     float64 
 5   body_mass_g        44 non-null     float64 
 6   sex                44 non-null     category
 7   year               44 non-null     category
 8   mass_kg            44 non-null     float64 
dtypes: category(4), float64(5)
memory usage: 2.7 KB


<h3>Insight 7</h3>
<p>Shown above are the result of Chinstrap and Gentoo penguins recorded from year 2009. There are 24 total of Chinstrap and 44 total of Gentoo penguin from year 2009. </p>
<hr>